src/cache/cache.rs

```diff
@@ -65,6 +65,17 @@ fn set_file_mode(_path: &Path, _mode: u32) -> Result<()> {
     Ok(())
 }
 
+/// Cache object sourced by a file.
+#[derive(Clone)]
+pub struct FileObjectSource {
+    /// Identifier for this object. Should be unique within a compilation unit.
+    pub key: String,
+    /// Path to the file.
```

Comment:
nit: specify if this is a absolute or relative path

Reply:
👍